From: Zi Yan <ziy@nvidia.com>
Date: Thu, 8 May 2025 15:23:42 -0400
Subject: [PATCH] 
 mm-page_isolation-remove-migratetype-from-move_freepages_block_isolate-fix

Here is the fixup 2/3 to address Johannes' comments.

1. replace get_pageblock_isolate() with
   get_pageblock_migratetype() == MIGRATE_ISOLATE, a
   get_pageblock_isolate() becomes private in mm/page_isolation.c

Link: https://lkml.kernel.org/r/F05CF961-6104-4C46-94C2-8966F580D8AC@nvidia.com
Signed-off-by: Zi Yan <ziy@nvidia.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/page_alloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index b502e4a3afbe..c99b77fdf383 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -1975,7 +1975,7 @@ static bool __move_freepages_block_isolate(struct zone *zone,
 	int from_mt;
 	int to_mt;
 
-	if (isolate == get_pageblock_isolate(page))
+	if (isolate == (get_pageblock_migratetype(page) == MIGRATE_ISOLATE))
 		return false;
 
 	if (!prep_move_freepages_block(zone, page, &start_pfn, NULL, NULL))
-- 
2.39.5

